### ISSUE TITLE
CLI config option stomps on --output param

### ISSUE THREAD

**Issue by DennisGCronin** [javascript-obfuscator/javascript-obfuscator#653]
The docs suggest that CLI options override config options but when I specify --config [jso].config it ignores the --output out.js which otherwise works fine.  Then it just spews all the obfuscated code to the console. If I omit the --config option that output is properly written to the spec'd file.

Running on Centos Linux.  Using this:

~/node_modules/javascript-obfuscator/bin/javascript-obfuscator in-src.js --config low-ob-high-perf.config --output out.js

The config file low-ob-high-perf.config is just a cut-n-paste of the suggested options from here under Preset Options:

https://github.com/javascript-obfuscator/javascript-obfuscator

Note: it might be useful to add a CLI param --perf-level [low|med|hi] or something like that with the preset options.

Thanks.

**Comment by sanex3339** [comment]
You should use `js` or `json` extension for the config file. I don't know why it works with the production bundle, it doesn't work with my dev bundle. Will add a check for the valid extensions.

**Comment by DennisGCronin** [comment]
Thanks for fast response! 

I tried the .js extension for the config and it still spews to the console. Just for grins I tried omitting the --output out.js hoping it would write to in-obfuscated.js but it still spews to the console.  So appears any config causes it to output to STDOUT. Curious.

But ya know, it looks like the default options pretty much give me the low-obfuscation-high-performance settings so I'm going to just go with no config.  Thanks.

Dennis

**Comment by sanex3339** [comment]
Show me content of your config file with JS extension

**Comment by DennisGCronin** [comment]
Just cribbing from here: https://github.com/javascript-obfuscator/javascript-obfuscator

For example, my low-ob-high-perf.js file contains:

```
{
    compact: true,
    controlFlowFlattening: false,
    deadCodeInjection: false,
    debugProtection: false,
    debugProtectionInterval: false,
    disableConsoleOutput: true,
    identifierNamesGenerator: 'hexadecimal',
    log: false,
    renameGlobals: false,
    rotateStringArray: true,
    selfDefending: true,
    shuffleStringArray: true,
    splitStrings: false,
    stringArray: true,
    stringArrayEncoding: false,
    stringArrayThreshold: 0.75,
    unicodeEscapeSequence: false
}

```

Running: node: 14.5.0

Like I said this is non-critical as I think I've determined the default options pretty much set up for low-obfuscation high-perf.

**Comment by sanex3339** [comment]
This not a valid exproted config. With THIS content of file, rename it to json file, or keep is as js file but export object with config with module.exports

**Comment by sanex3339** [comment]
Additional check released as `1.3.0`

**Comment by DennisGCronin** [comment]
Ah ha, that config needs to be valid JSON.  Didn't even think about it, just cut-n-paste from that page.

Making it valid JSON *and* using .json suffix makes it work.  Otherwise the big source barf up is not my source, it's apparently the obfuscator source.

And the stack trace is:

```
ReferenceError: Given config path must be a valid `.js` or `.json` file path
    at Function.getUserConfig (/home/denny/node_modules/javascript-obfuscator/dist/index.cli.js:24:431466)
    at Function.buildOptions (/home/denny/node_modules/javascript-obfuscator/dist/index.cli.js:24:420605)
    at F.initialize (/home/denny/node_modules/javascript-obfuscator/dist/index.cli.js:24:420961)
    at F.value (/home/denny/node_modules/javascript-obfuscator/dist/index.cli.js:24:11871)
    at F.value (/home/denny/node_modules/javascript-obfuscator/dist/index.cli.js:24:11871)
    at F.value (/home/denny/node_modules/javascript-obfuscator/dist/index.cli.js:24:11871)
    at F.value (/home/denny/node_modules/javascript-obfuscator/dist/index.cli.js:24:11871)
    at F.value (/home/denny/node_modules/javascript-obfuscator/dist/index.cli.js:24:11871)
    at Function.obfuscate (/home/denny/node_modules/javascript-obfuscator/dist/index.cli.js:24:429390)
    at Object.<anonymous> (/home/denny/node_modules/javascript-obfuscator/bin/javascript-obfuscator:3:30)
```

If the input .js is not valid, same dealie, big source barf up to stderr.

It's working now but it would be cool if instead the program said:

Error: invalid Javascript input file

or:

Error: config file is not valid JSON with .json suffix

Or something like that.  Anyhow, thanks for your help. 

Dennis
